### PR TITLE
have a common interface to extract metadata from SizeNodes

### PR DESCRIPTION
**Pull request by Krovatkin**
Fixes #ISSUE_NUMBER

### CODE REVIEW

File: torch/csrc/lazy/ts_backend/dynamic_ir.h [hunk 1]
```diff
@@ -43,20 +44,8 @@ namespace lazy {
...
-  }
-
-  std::string ToString() const override;
-
-  virtual int64_t getStaticValue() const = 0;
-};
-
 // Represents the result of calling `size` on a Tensor
-class TORCH_API SizeNode : public DimensionNode {
+class TORCH_API SizeNode : public TsNode, public DimensionNode {
```
[the remark this answers is not in the document]
Comment (Krovatkin) [reply torch/csrc/lazy/ts_backend/dynamic_ir.h:48]: Correct! I think I should just add a test. Initially, I just wanted to get @miladm to try it out w/ his PR :)

[the remark this answers is not in the document]
Comment (Krovatkin) [reply torch/csrc/lazy/ts_backend/dynamic_ir.h:48]: added a test

**Review by JackCaoG — APPROVED** [review]
Thanks!

**Review by wconstab — APPROVED** [review]
lgtm, thanks nick!

File: test/cpp/lazy/test_ir.cpp [hunk 2]
```diff
@@ -9,6 +9,7 @@
 #include <torch/csrc/lazy/core/ir_metadata.h>
 #include <torch/csrc/lazy/ts_backend/ts_node.h>
 #include <memory>
+#include "torch/csrc/lazy/core/dynamic_ir.h"
```
Comment (Gamrix) [inline test/cpp/lazy/test_ir.cpp:12]: I thought we are supposed to use `<>` inports.

Comment (Krovatkin) [reply test/cpp/lazy/test_ir.cpp:12]: yes we are... that's clang-tidy :(

**Comment by miladm** [comment]
FIXES https://github.com/pytorch/pytorch/issues/77880

**Pull request closed without merging**